src/panoptes/pocs/focuser/focuser.py

```diff
@@ -470,100 +470,68 @@ def _autofocus(self,
             # Fit models to data
             fit = fitter(reparameterised_polynomial,
                          focus_positions[fitting_indices[0]:fitting_indices[1] + 1],
-                         metric[fitting_indices[0]:fitting_indices[1] + 1])
+                         metrics[fitting_indices[0]:fitting_indices[1] + 1])
 
-            best_focus = -fit.offset_0
+            # Get the encoder position of the best focus.
+            best_focus = np.abs(fit.offset_0)
             fitted = True
 
-            # Guard against fitting failures, force best focus to stay within sweep range
+            # Guard against fitting failures, force best focus to stay within sweep range.
             min_focus = focus_positions[0]
             max_focus = focus_positions[-1]
             if best_focus < min_focus:
-                self.logger.warning("Fitting failure: best focus {} below sweep limit {}",
-                                    best_focus,
-                                    min_focus)
-
+                self.logger.warning(f"Fitting failure: best focus {best_focus} below sweep limit {min_focus}")
                 best_focus = focus_positions[1]
 
             if best_focus > max_focus:
-                self.logger.warning("Fitting failure: best focus {} above sweep limit {}",
-                                    best_focus,
-                                    max_focus)
-
+                self.logger.warning(f"Fitting failure: best focus {best_focus} above sweep limit {max_focus}")
                 best_focus = focus_positions[-2]
 
         else:
             # Coarse focus, just use max value.
             best_focus = focus_positions[imax]
 
+        # Move the focuser to best focus position.
         final_focus = self.move_to(best_focus)
 
-        final_fn = f"{final_focus}_{focus_type}_final.{self._camera.file_extension}"
+        # Get final thumbnail.
+        final_fn = f"{final_focus}-{focus_type}-final.{self._camera.file_extension}"
         file_path = os.path.join(file_path_root, final_fn)
         try:
-            final_thumbnail = self._camera.get_thumbnail(
-                seconds, file_path, thumbnail_size, keep_file=True)
+            final_thumbnail = self._camera.get_thumbnail(seconds, file_path, thumbnail_size, keep_file=True)
+            if dark_thumbnail is not None:
+                final_thumbnail = final_thumbnail - dark_thumbnail
```

Comment:
Potential integer underflow.

Reply:
Bound to be underflow.
```suggestion
                final_thumbnail = final_thumbnail.astype(np.int32) - dark_thumbnail
```